Session-2.0.kt

```diff
@@ -0,0 +1,50 @@
+import java.util.ArrayList
+
+fun main(args: Array<String>) {
+
+    val streetCap = 8; // Capacity of street
+    val countOfAllCars = 25; //Number of cars
+    var drivingCars: Int = 0 // Number of cars that want to drive
+    var allCars = ArrayList<Car>(); // Array of cars
+
+
+    for (i in 0..countOfAllCars - 1) {
+        try { //Exception handling
+            allCars.add(Car())
+        } catch (e: Exception) {
+            //do something usefull or useless
+        }
+
+    }
+
+    var driveIt: Boolean
+    for (i in 0..countOfAllCars - 1) {    // Count the number of driving cars
+        driveIt = allCars[i].isDriving()
+        if (driveIt) {
+            drivingCars ++
+        }
+    }
+    println(drivingCars.toString() + " cars are driving! ") //Printnumber of cars that want to drive
+    if (drivingCars <= streetCap) { // check if they are delayed or not
+        print("all can drive")
+    } else {
+        print("all are delayed")
+    }
+
+}
+class Car {
+    internal var amDriving: Boolean = false //is this car driving now?
+
+    fun isDriving(): Boolean {
+        val value = Math.random() //random value between 0 and 1

```

Comment:
The desire to drive should not be randomized for each request. If the simulation would ask twoce whether a car wanted to drive (e.g. during simulation and during result printing), it might get different answers. Instead please pass the desire to drive via the constructor.

If you still want to have a random number of driving cars (which makes sense), randomize the number of cars or their desire to drive during construction.